go/fn/const.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package fn
+
+const (
+	// internalPrefix is the prefix given to internal annotations that are used
+	// internally by the orchestrator
+	internalPrefix string = "internal.config.kubernetes.io/"
+
+	// IndexAnnotation records the index of a specific resource in a file or input stream.
+	IndexAnnotation string = internalPrefix + "index"
```

Comment:
Since all these consts are already defined in kyaml, should we instead define them as `IndexAnnotation string = kioutil.IndexAnnotation`? That could help us keep it consistent with upstream kyaml.

Reply:
This constants are not kustomize specific and we do not expect users to import kyaml lib when using the SDK. Thus, it's better to keep it separate from kustomize/kyaml